consumer/consumererror/signalerrors.go

```diff
@@ -24,20 +33,44 @@ func (err retryable[V]) Data() V {
 	return err.data
 }
 
+// Delay returns the time duration before the telemetry data should
+// be resent to the destination.
+func (err retryable[V]) Delay() time.Duration {
+	return err.delay
+}
+
+// RetryOption allows adding data to a retryable error,
+// e.g. the duration before sending should be retried.
+type RetryOption func(err *retryableCommon)
+
+func WithRetryDelay(delay time.Duration) RetryOption {
```

Comment:
Should instead have a standalone `RetryDelay` error?

Reply:
We should probably keep a single error for a few reasons:

1. We agreed that upstream components should use the copy of data returned by the downstream component, so delays without data would be inconsistent with that.
2. Upstream components can get both the data and retry information out of the error a bit more easily.
3. These two concepts are pretty closely related; if you want to retry data, you likely also care about how long to wait before retrying it.